topdown/builtins.go

```diff
@@ -80,15 +80,22 @@ func (BuiltinEmpty) Error() string {
 
 var builtinFunctions = map[string]BuiltinFunc{}
 
+func builtinWrapper(name string, fn BuiltinFunc) BuiltinFunc {
+	return func(bctx BuiltinContext, args []*ast.Term, iter func(*ast.Term) error) error {
+		err := fn(bctx, args, iter)
+		if err == nil {
+			return nil
+		}
+		return handleBuiltinErr(name, bctx.Location, err)
+	}
+}
+
 func functionalWrapper1(name string, fn FunctionalBuiltin1) BuiltinFunc {
 	return func(bctx BuiltinContext, args []*ast.Term, iter func(*ast.Term) error) error {
 		result, err := fn(args[0].Value)
 		if err == nil {
 			return iter(ast.NewTerm(result))
 		}
-		if _, empty := err.(BuiltinEmpty); empty {
-			return nil
-		}
```

Comment:
Was this change (and the related ones below) supposed to go in? Built-ins implementing the "functional" interface used to return this to indicate they were undefined. There are still a few type-related and aggregate functions doing this.

Reply:
Yea.. its removing duplicate code. The `handleBuiltinErr` already does this same thing down on line 138 (unless I'm missing some subtlety with the original):

```
func handleBuiltinErr(name string, loc *ast.Location, err error) error {
	switch err := err.(type) {
	case BuiltinEmpty:
		return nil
``` 
So it seemed weird to explicitly check this, then call the the handle err and do a type switch which did the same check and returns nil.

Reply:
@patrick-east thanks for clarifying! I was only paying attention to the diff and forgot about the other place where we're checking for the BuiltinEmpty type.